sips/sip-16/sip-016-token-metadata.md

```diff
@@ -0,0 +1,394 @@
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Token Metadata Property",
+    "type": "object",
+    "required": [],
+    "properties": {
+        "type": {
+            "type": "string",
+            "description": "type of the property"
+        },
+        "description": {
+            "type": "string",
+            "description": "description of the property"
+        },
+        "value": {
+            "type": {"oneOf": [{"type": "object"}, {"type": "string"}, {"type": "number"}, {"type": "integer"}, {"type": "boolean"} {"type: "array"}},
+            "description": "value of the property"
+        }
+    }
+}
+```
+
+Example:
+
+```
+{
+    "type": "string",
+    "description": "Address of custodian key holder",
+    "value": "Casa Inc., P.O. Box 20575, Charleston, S.C. 29413, United States."
+}
+```
+
+### Attributes
+
+Attributes describe addition elements of tokens that are "observable", usually represented in the image of the token.
+
+In contrast, properties describe elements of tokens that are more abstract.
+
+An attribute consists of a `trait_type` defining the name of the trait, `value` is the value of the trait, and `display_type` is a field indicating how you would like it to be displayed.
+
+Appendix B describes type of attributes
+
+## Localization
+
+The localized data follow the same JSON schema with property `sip` as required and all other properties as optional.
+
+The localized data overwrite data provided in the default meta data JSON. The localized data can provide only partial data.
+
+An array of localized `attributes` overwrites the whole list of default `attributes`.
+
+A localized properties with partial data overwrites only the provided properties, the remaining default properties remain as default values.
+
+# Using metadata in applications
+
+Before presenting metadata to users, application developers should verify whether the metadata is compliant with the application's guidelines.
+
+We remind implementation authors that the empty string for the token uri is a valid response. We also remind everyone that any smart contract can use the same metadata as other NFT contracts. It is out of the scope of this standard to define how a client may determine which smart contracts are the original, well-known, canonical ones.
+
+## Graphical representation
+
+The metadata of a token contain several properties that can be used to visually represent the token. It is recommended to use the first defined property of the following list:
+
+1. `image`
+2. `properties.image_data`
+3. `image_data`
+
+Rich representation should use ther first defined property of the following list:
+
+1. `properties.animation_url`
+2. `animation_url`
+
+# Out of Scope
+
+Accessiblity of content is not covered by the standard.
+
+Properties other than resolvability of the token uri are out of scope. This implies that metadata might change over time (stability).
+
+# Metadata functions
+
```

Comment:
Can this section speak more to the extent that metadata functions should be relied upon, now that this SIP exists?  For example, does this SIP supersede all metadata functions?  Does this SIP require that certain fields have the same values as those returned by corresponding metadata functions?  I'm not sure how to relate these two things.

Reply:
I tried to improve this section.

Reply:
Thanks! Looks good.